src/test/integration/modules/subject/subject-service.spec.js

```diff
@@ -0,0 +1,104 @@
+const mongoose = require('mongoose')
+const { MongoMemoryServer } = require('mongodb-memory-server')
+const subjectService = require('./subject.service')
+const Subject = require('./subject.model')
+
+let mongoServer
+
+beforeAll(async () => {
+  mongoServer = await MongoMemoryServer.create()
+  const uri = mongoServer.getUri()
+  await mongoose.connect(uri, { useNewUrlParser: true, useUnifiedTopology: true })
```

Comment:
Good approach using `mongodb-memory-server`. Consider creating a task to set up all tests with this approach. I mean updating `src/test/setup.js`.